### PR TITLE
Add tag protection

### PR DESCRIPTION
**Pull request by jfengupgrade**

### CODE REVIEW

File: example/tagprotection/main.go
Comment (gmlewis) [inline example/tagprotection/main.go:7]:
```suggestion
// prompts the user for GitHub owner, repo, tag protection pattern and token,
```

File: example/tagprotection/main.go
Comment (gmlewis) [inline example/tagprotection/main.go:8]:
```suggestion
// then creates a new tag protection if the user entered a pattern at the prompt.
```

File: example/tagprotection/main.go
Comment (gmlewis) [inline example/tagprotection/main.go:9]:
```suggestion
// Otherwise, it will just list all existing tag protections.
```

File: example/tagprotection/main.go
Comment (gmlewis) [inline example/tagprotection/main.go:56]:
```suggestion
		tagProtection, _, err := client.Repositories.CreateTagProtection(ctx, owner, repo, pattern)
```

File: example/tagprotection/main.go
Comment (gmlewis) [inline example/tagprotection/main.go:70]:
```suggestion
	tagProtections, _, err := client.Repositories.ListTagProtection(ctx, owner, repo)
```

File: example/tagprotection/main.go
Comment (gmlewis) [inline example/tagprotection/main.go:59]: Instead of `fmt.Printf()... return`, let's please just delete the return and use `log.Fatalf` instead.  
Here, and below.

File: example/tagprotection/main.go
Comment (gmlewis) [inline example/tagprotection/main.go:73]: Here too.

File: github/repos_tags.go
Comment (gmlewis) [inline github/repos_tags.go:19]:
```suggestion
// tagProtectionRequest represents a request to create tag protection.
```

File: github/repos_tags.go
Comment (gmlewis) [inline github/repos_tags.go:20]: Let's unexport this struct since it is only used internally.

```suggestion
type tagProtectionRequest struct {
```

File: github/repos_tags.go
Comment (gmlewis) [inline github/repos_tags.go]:
```suggestion
func (s *RepositoriesService) DeleteTagProtection(ctx context.Context, owner, repo string, tagProtectionID int64) (*Response, error) {
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @jfengupgrade !
Just a few tweaks, please.

**Review by gmlewis — APPROVED** [review]
Thank you, @jfengupgrade !
LGTM.

Awaiting second LGTM from any other contributor to this repo before merging.

**Comment by jfengupgrade** [comment]
@gmlewis Thanks a lot for the reviewing

**Review by raynigon — APPROVED** [review]
LGTM

**Comment by gmlewis** [comment]
Thank you, @raynigon !
Merging.

**Pull request merged**
